.github/scripts/generate_pytorch_test_matrix.py

```diff
@@ -9,22 +9,47 @@
 
 import json
 import os
-from typing import List
+from typing import Dict
 
+from typing_extensions import TypedDict
 
-NUM_TEST_SHARDS = int(os.getenv('NUM_TEST_SHARDS', '1'))
 
-def generate_sharding_list() -> List[int]:
-    return list(range(1, NUM_TEST_SHARDS + 1))
+class Config(TypedDict):
+    num_shards: int
+    runner: str
 
 
 def main() -> None:
-    print(json.dumps(
-        {
-            'test_config': generate_sharding_list()
-        },
-        sort_keys=True,
-    ))
+    TEST_RUNNER_TYPE = os.getenv('TEST_RUNNER_TYPE')
```

Comment:
Instead of adding all of this logic here why not take a simpler approach and just make the multi-gpu job a separate job that gets included in the jinja2 generation if `enable_multigpu_test` is True.

Similar to what we do with the docs build, just so we're not introducing complexity where we don't actually need to. It won't be exactly DRY but at least it'll be a bit easier to understand

Reply:
@seemethere Ok that sounds fine to me! I'll go ahead and do that

Reply:
Talked with @samestep and we agree that this approach is fine